### PR TITLE
Upgrade/masonite 4

### PR DESCRIPTION
**Pull request by josephmancuso**

### CODE REVIEW

**Comment by josephmancuso** [comment]
@waghanza we should probably delete whatever was on master and just merge this new project

**Comment by waghanza** [comment]
@josephmancuso  When requesting `/`, I have
```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/masonite/foundation/Application.py", line 49, in __call__
    return self.response_handler(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/masonite/foundation/response_handler.py", line 29, in response_handler
    application.make("exception_handler").handle(e)
  File "/usr/local/lib/python3.9/site-packages/masonite/exceptions/ExceptionHandler.py", line 39, in handle
    self.application.make("event").fire(
  File "/usr/local/lib/python3.9/site-packages/masonite/container/container.py", line 105, in make
    raise MissingContainerBindingNotFound(
masonite.exceptions.exceptions.MissingContainerBindingNotFound: event key was not found in the container
```

File: python/masonite/Kernel.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+from masonite.auth import Sign
+from masonite.environment import LoadEnvironment
+from masonite.utils.structures import load
+from masonite.utils.location import base_path
+from masonite.middleware import (
+    SessionMiddleware,
+    EncryptCookies,
+    LoadUserMiddleware,
+    MaintenanceModeMiddleware,
+    VerifyCsrfToken,
```
Comment (waghanza) [inline python/masonite/Kernel.py:12]: initially the version inside this implementation was used, why not using the one provided by `masonite`

**Review by waghanza — CHANGES_REQUESTED** [review]

File: python/masonite/Kernel.py (hunk 1 above)
Comment (josephmancuso) [reply python/masonite/Kernel.py:12]: It is basically. This class is in new projects and inherits the Masonite class basically just for the purpose of being able to exclude routes from csrf

**Comment by josephmancuso** [comment]
I don't get that issue when I run the server locally .. what are you doing?

The issue here is when a new exception is thrown it fires an event ... but I removed the event provider because there shouldn't be an exception thrown anyway. 

Even if I add it back it will return an exception but idk why it's throwing exceptions on that route ...

**Comment by waghanza** [comment]
I run `gunicorn wsgi`, how can I run `masonite` otherwise ?

1. I run into the container with `bundle exec rake config && docker build -f .Dockerfile  -t test . && docker run -p 3000:3000 -ti test bash`
2. I start `gunicorn` with `gunicorn --bind 0.0.0.0:3000 wsgi`
3. I send an http request to `/` with `curl http://127.0.0.1:3000 -v` (since my container IP is bound to 127.0.0.1)

**Review by waghanza — APPROVED** [review]

**Pull request merged**